src/server/services/pickup-queue/socket-methods.js

```diff
@@ -51,9 +76,13 @@ export default function socketMethods(app, socket) {
         preReady: null,
       });
 
-      log('Adding user to pickup', userId);
+      if (await checkBlock(app, userId)) {
+        log('User blocked for pickup', userId);
```

Comment:
Emit a socket.io event here that the user can't join the event

Example: https://github.com/TF2PickupNET/TF2Pickup/blob/dev/src/server/services/pickup-queue/statuses/ready-up/create-pickup.js#L123